### ISSUE TITLE
Postgres on Ubuntu Replication Issue

### ISSUE THREAD

**Issue by stevenlivz** [Azure/azure-quickstart-templates#2103]
I feel I may be missing something but I had been using the following repo:

https://github.com/Azure/azure-quickstart-templates/tree/master/postgresql-on-ubuntu

I followed all the instructions but couldn't get streaming replication working. It took me a few days as i expected it was something I was doing wrong with permissions but it *seems* that you do not set the data directory after you have done all your configuration in setup_datadisks.

As soon as I set the data_directory in postgressql.conf in /etc/postgresql/9.3/main to point at the new data directory it all started working.

However, this feels very basic as it simply wouldn't work without this being set so i can't believe I am the first to find it .... is this really a bug or have i done something that would have caused this not work work?

thanks,
steven

**Comment by tstringer** [comment]
I'm seeing the same thing. @stevenlivz did you modify `postgresql.conf` *after* the fact, or did you have to modify the `install_postgres.sh` script? I take it it's the former. I setup logging for this custom script, and here is the output I get:

```
mv: cannot stat '/var/lib/kafkadir': No such file or directory
could not change directory to "/var/lib/waagent/custom-script/download/0": Permission denied
```

**Comment by stevenlivz** [comment]
@tstringer Yeah, after the fact. I cloned the repo and modified the configure_streaming_replication method in install_postgresql.sh as follows:

`configure_streaming_replication() {
	logger "Starting configuring PostgreSQL streaming replication..."

	# Configure the MASTER node
	if [ "$NODETYPE" == "MASTER" ];
	then
		logger "Create user replicator..."
		echo "CREATE USER replicator WITH REPLICATION PASSWORD '$PGPASSWORD';"
		sudo -u postgres psql -c "CREATE USER replicator WITH REPLICATION PASSWORD '$PGPASSWORD';"

		### added - move the data file to the new mount point
		sudo mv /var/lib/postgresql/9.3/main /var/lib/kafkadir      ### added
	fi ...`

**Comment by jomonmavila** [comment]
How can I use existing virtual network into this. There is a lot of dependencies there.
